lib/backend.js

```diff
@@ -810,6 +810,7 @@ Backend.prototype.fetchSnapshotByTimestamp = function(agent, index, id, timestam
     id: id,
     timestamp: timestamp
   };
+  timestamp = timestamp ? timestamp : Date.now();
```

Comment:
Also, while this is a nice quick fix, I do wonder if we should address the deeper issues here (ie that `null` should be able to be passed through each step of this machinery).

Namely, if I pass `null` to `getMilestoneSnapshotAtOrAfterTime()` should it actually return the live snapshot (rather than the most recent milestone?).

Alternatively we could insert some short-circuit logic here for a performance gain: maybe now is finally the time to just check if we're asking for a `null` timestamp or version, and just directly call `db.getSnapshot()`? That would potentially save us a couple of DB round-trips, which feels like a nice win, especially if consumers are just using `fetchSnapshot()` as a convenience method for fetching the latest snapshot (which we are!).

Let's discuss with @ericyhwang next week.

Reply:
> Namely, if I pass null to `getMilestoneSnapshotAtOrAfterTime()` should it actually return the live snapshot (rather than the most recent milestone?).

As discussed, the milestone snapshot adapter is separate from the normal db adapter - the Mongo version uses [a different collection](https://github.com/share/sharedb-milestone-mongo/blob/6cf429ed4a1ccb5f02a016b304d9a928a381fed5/lib/mongo-milestone-db.js#L210-L212) - so we can't return the latest actual snapshot in this case.

> Alternatively we could insert some short-circuit logic here for a performance gain: maybe now is finally the time to just check if we're asking for a null timestamp or version, and just directly call `db.getSnapshot()`?

Yeah, makes sense, for both fetchSnapshot by version and timestamp